webapp/tests/karma/unit/controllers/inbox.js

```diff
@@ -180,6 +182,15 @@ describe('InboxCtrl controller', () => {
     chai.expect(session.init.callCount).to.equal(2);
   });
 
+  it('Tour modal should not be displayed if no tours are availbable', done => {
+    const controller = createController();
+    setTimeout(() => {
+      chai.expect(controller.filteredModals.length).to.equal(0);
+      done();
+    });
+    scope.$digest();
+  });
+
```

Comment:
Nice one! Just to prove it works as expected, can you also write a test that asserts the `filtereredModals` is 1 when there _is_ a tour available?

Reply:
Good catch, this had fooled me -- filteredModals (now modalsToShow) was made 0 since shift() was being called within showModals, so the length would never have been 1.  I modified the test to determine whether or not render was actually called.  This included nesting another setTimeout, I'm not sure if there is a cleaner way to do this or not.